tests/integration/ambient/waypoint_test.go

```diff
@@ -61,3 +67,131 @@ func TestWaypointStatus(t *testing.T) {
+				t.Fatal(err)
+			}
+
+			istioctl.NewOrFail(t, t, istioctl.Config{}).InvokeOrFail(t, []string{
+				"x",
+				"waypoint",
+				"apply",
+				"--namespace",
+				nsConfig.Name(),
+			})
+			retry.UntilSuccessOrFail(t, func() error {
+				fetch := kubetest.NewPodFetch(t.AllClusters()[0], nsConfig.Name(), constants.GatewayNameLabel+"="+"namespace")
+				if _, err := kubetest.CheckPodsAreReady(fetch); err != nil {
+					return fmt.Errorf("gateway is not ready: %v", err)
+				}
+				return nil
+			}, retry.Timeout(15*time.Second), retry.BackoffDelay(time.Millisecond*100))
+
+			saSet := []string{"sa1", "sa2", "sa3"}
+			for _, sa := range []string{"sa1", "sa2", "sa3"} {
+				istioctl.NewOrFail(t, t, istioctl.Config{}).InvokeOrFail(t, []string{
+					"x",
+					"waypoint",
+					"apply",
+					"--namespace",
+					nsConfig.Name(),
+					"--service-account",
+					sa,
+				})
+				retry.UntilSuccessOrFail(t, func() error {
+					fetch := kubetest.NewPodFetch(t.AllClusters()[0], nsConfig.Name(), constants.GatewayNameLabel+"="+sa)
+					if _, err := kubetest.CheckPodsAreReady(fetch); err != nil {
+						return fmt.Errorf("gateway is not ready: %v", err)
+					}
+					return nil
+				}, retry.Timeout(15*time.Second), retry.BackoffDelay(time.Millisecond*100))
+			}
+
+			output, _ := istioctl.NewOrFail(t, t, istioctl.Config{}).InvokeOrFail(t, []string{
+				"x",
+				"waypoint",
+				"list",
+				"--namespace",
+				nsConfig.Name(),
+			})
+			for _, sa := range saSet {
+				if !strings.Contains(output, sa) {
+					t.Fatalf("expect to find %s in output: %s", sa, output)
+				}
+			}
+
+			output, _ = istioctl.NewOrFail(t, t, istioctl.Config{}).InvokeOrFail(t, []string{
+				"x",
+				"waypoint",
+				"list",
+				"-A",
+			})
+			for _, sa := range saSet {
+				if !strings.Contains(output, sa) {
+					t.Fatalf("expect to find %s in output: %s", sa, output)
+				}
+			}
+
+			istioctl.NewOrFail(t, t, istioctl.Config{}).InvokeOrFail(t, []string{
+				"x",
+				"waypoint",
+				"-n",
+				nsConfig.Name(),
+				"delete",
+			})
+			retry.UntilSuccessOrFail(t, func() error {
+				fetch := kubetest.NewPodFetch(t.AllClusters()[0], nsConfig.Name(), constants.GatewayNameLabel+"="+"namespace")
+				// after deletion, the pod should not be found
+				if _, err := kubetest.CheckPodsAreReady(fetch); err != nil {
+					if errors.Is(err, kubetest.ErrNoPodsFetched) {
+						return nil
+					}
+					return fmt.Errorf("failed to check gateway status: %v", err)
+				}
+				return fmt.Errorf("failed to clean up gateway in namespace: %s", nsConfig.Name())
+			}, retry.Timeout(15*time.Second), retry.BackoffDelay(time.Millisecond*100))
+
+			istioctl.NewOrFail(t, t, istioctl.Config{}).InvokeOrFail(t, []string{
+				"x",
+				"waypoint",
+				"-n",
+				nsConfig.Name(),
+				"delete",
+				"sa1",
+				"sa2",
+			})
+			retry.UntilSuccessOrFail(t, func() error {
+				fetch := kubetest.NewPodFetch(t.AllClusters()[0], nsConfig.Name(), constants.GatewayNameLabel+"="+"sa1")
```

Comment:
again, code duplication. This check happens twice, with the only difference being the selector label.